src/basic/fileio.c

```diff
@@ -80,6 +80,18 @@ FILE * take_fdopen(int *fd, const char *options) {
         return f;
 }
 
+DIR * take_fdopendir(int *dfd) {
```

Comment:
`DIR* ` would be nicer.

Reply:
Oops, fixed, for take_fdopen() in the .c file too.